### PR TITLE
Add option to skip tests after retries

### PR DESCRIPTION
**Pull request by malfet**

### CODE REVIEW

File: torch/testing/_internal/common_utils.py [hunk 1]
```diff
@@ -1298,7 +1298,7 @@ def wrapper(*args, **kwargs):
 
 
 # Decorator to retry upon certain Exceptions.
-def retry(ExceptionToCheck, tries=3, delay=3):
+def retry(ExceptionToCheck, tries=3, delay=3, skip_after_retires=True):
```
Comment (mruberry) [inline torch/testing/_internal/common_utils.py:1301]: retires->retries

**Review by ailzhang — COMMENTED** [review]
Thanks!

File: torch/testing/_internal/common_utils.py [hunk 2]
```diff
@@ -1311,7 +1311,10 @@ def f_retry(*args, **kwargs):
                     print(msg)
                     time.sleep(mdelay)
                     mtries -= 1
-            return f(*args, **kwargs)
+            try:
+                return f(*args, **kwargs)
+            except ExceptionToCheck as e:
+                raise unittest.SkipTest(f"Skipping after {tries} consequentive {str(e)}") if skip_after_retires else e
```
Comment (mruberry) [inline torch/testing/_internal/common_utils.py:1317]: consequentive->consecutive?

File: torch/testing/_internal/common_utils.py (hunk 1 above)
Comment (malfet) [reply torch/testing/_internal/common_utils.py:1301]: Corrected

File: torch/testing/_internal/common_utils.py (hunk 2 above)
Comment (malfet) [reply torch/testing/_internal/common_utils.py]: OMG, sorry, what's wrong with me today

**Review by mruberry — APPROVED** [review]
Awesome!

**Pull request closed without merging**